docs/tutorials/13-diarization.md

```diff
@@ -0,0 +1,292 @@
+
+```python
+class Segment:
+    def __init__(self, start, end, speaker=None):
+        self.start = start
+        self.end = end
+        self.speaker = speaker
+```
+
+**Assigning Speakers**
+
+This is the process that involves taking the transcribed chunks and assigning them to the speakers discovered by the Speaker Diarization Pipeline.
+
+In this function, timestamps of the different chunks are compared against the start and end times of speakers in the DataFrame returned by the `SpeakerDiarization` pipeline segments of a transcript are assigned speaker labels based on the overlap between the speech segments and diarization data.
+
+The function iterates through segments of a transcript and assigns the speaker labels based on the overlap between the speech segments and the diarization data.
+
+In case of no overlap, a the fill_nearest parameter can be set to `True`, then the function will assign the speakers to segments by finding the closest speaker in time.
+
+The function takes parameters:
+    
+`diarize_df`: a pandas DataFrame returned by the DiarizationPipeline containing the diarization information with columns like `start`, `end` and `speaker`
+
+`transcript_result`: A dictionary with a key `chunks` that contains a list of trancript `Segments` obtained from the ASR pipeline.
+
+`fill_nearest`: Default is `False`
+
+`Returns:` An updated `transcript_result` with speakers assigned to each segment in the form:
+
+```python
+{
+    'text':'Hello World',
+    'chunks':[
+        {'text': 'Hello', 'timestamp': (0.5, 1.0), 'speaker': 0},
+        {'text': 'world', 'timestamp': (1.5, 2.0), 'speaker': 1}]}
+```
+
+```python
+
+def assign_word_speakers(diarize_df, transcript_result, fill_nearest=False):
+ 
+    transcript_segments = transcript_result["chunks"]
+
+    for seg in transcript_segments:
+        # Calculate intersection and union between diarization segments and transcript segment
+        diarize_df['intersection'] = np.minimum(diarize_df['end'], seg["timestamp"][1]) - np.maximum(diarize_df['start'], seg["timestamp"][0])
+        diarize_df['union'] = np.maximum(diarize_df['end'], seg["timestamp"][1]) - np.minimum(diarize_df['start'], seg["timestamp"][0])
+
+        # Filter out diarization segments with no overlap if fill_nearest is False
+        if not fill_nearest:
+            dia_tmp = diarize_df[diarize_df['intersection'] > 0]
+        else:
+            dia_tmp = diarize_df
+
+        # If there are overlapping segments, assign the speaker with the greatest overlap
+        if len(dia_tmp) > 0:
+            speaker = dia_tmp.groupby("speaker")["intersection"].sum().sort_values(ascending=False).index[0]
+            seg["speaker"] = speaker
+
+    return transcript_result
+
+```
+
+**Running the diarization model**
+```python
+diarize_model = DiarizationPipeline(use_auth_token=hf_token, device=device)
+diarize_segments = diarize_model("/content/kitakas_eng.mp3", min_speakers=1, max_speakers=2)
+
+diarize_segments
+```
```

Comment:
could we show sample outputs of what diarization looks like after this cell?

Reply:
Thank you for the feedback.

We have added a few sample outputs at the end of the documentation, one for after `diarize segments` and another for after `assign_word_speakers`

Reply:
Looks great now, will proceed to Merge this PR 
🚀